actionmetadata_test.go

```diff
@@ -122,3 +135,30 @@ func TestKnowledgeBase(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func doesActionRepoExist(filePath string) *github.Response {
+	var Branch string = "master"
+	splitOnSlash := strings.Split(filePath, "/")
+	owner := splitOnSlash[1]
+	repo := splitOnSlash[2]
+
+	PAT := os.Getenv("PAT")
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: PAT},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+
+	_, res, err := client.Git.GetRef(context.Background(), owner, repo, fmt.Sprintf("tags/%s", Branch))
```

Comment:
You need to use `client.Repository.Get` or something like that. You need to check if the repo exists or not.

Reply:
@varunsh-coder sir by using this method I wasn't able to validate the folder inside the repo, I even used the `get.ref` but that also didn't work.

So I used `client.Repositories.GetContents` this method to validate the path to `action.yml`.
I have tested it on my local machine with slight modification in repository name `42Crunch/api-security-audit-action` 
->`42Crunch/aapi-security-audit-action` and name of folder inside repo ` snyk/actions/docker/` -> ` snyk/actions/dockers/` using my PAT.

Here are the results:
![image](https://user-images.githubusercontent.com/73299843/151332099-2f9c7b21-6484-45d6-b87b-df8099496e5a.png)

Reply:
@Devils-Knight added PAT to the workflow. Can you please rebase so you get the latest test.yml workflow?